### PR TITLE
feat(localforger): add protocol hash to the constructor of LocalForger

### PR DESCRIPTION
**Pull request by roxaneletourneau**
In preparation for Ithaca support, where some operation schema has changed (ie endorsement), the protocol hash will be passed to the constructor of the LocalForger in order to select the right encoder, decoder.

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: packages/taquito-local-forging/src/taquito-local-forging.ts [hunk 1]
```diff
@@ -25,10 +26,10 @@ export function getCodec(codec: CODEC) {
   };
 }
 
-export { VERSION } from './version';
-
 export class LocalForger implements Forger {
-  private codec = getCodec(CODEC.MANAGER);
+  constructor(public readonly protocolHash = ProtocolsHash.PtHangz2) {}
```
Comment (dsawali) [inline packages/taquito-local-forging/src/taquito-local-forging.ts:30]: will this be something we'd change in the future in accordance to protocol updates?

Comment (roxaneletourneau) [reply packages/taquito-local-forging/src/taquito-local-forging.ts:30]: Yes, we will need to update the default value to match new protocol hash

Comment (dsawali) [reply packages/taquito-local-forging/src/taquito-local-forging.ts:30]: maybe passing a config that is named `currentProtocol` or something like that would be more descriptive, and instead we can just modify the value of `currentProtocol` and not pass them directly to the constructor

Comment (roxaneletourneau) [reply packages/taquito-local-forging/src/taquito-local-forging.ts:30]: I thought about it a little more, and I think that I will change my precedent response. The schema of endorsement is different in hangzhou and ithaca. So at the moment, we need to know which protocol is used to forge this kind of operation properly. The distinction is needed until the migration from hangzhou to ithaca on mainnet. After that, it won't really make sense to forge using the old schema and we should remove the distinction to keep the code cleaner.

**Pull request merged**